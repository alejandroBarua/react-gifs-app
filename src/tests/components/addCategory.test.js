import React from 'react';
import { shallow } from 'enzyme'
import AddCategory from '../../components/AddCategory';

describe('add category', () => {

	const setCategories = jest.fn(); // funcion 
	let wrapper =  shallow(<AddCategory setCategories={setCategories} />);

	beforeEach(() => { // para reiniciar el componente en cada test
		jest.clearAllMocks();
		wrapper = shallow(<AddCategory setCategories={setCategories} />);
	})

	
	test('should be displayed correctly', () => {

		expect(wrapper).toMatchSnapshot();
	})

	test('value to change input', () => {

		const value = 'Hello World';

		wrapper.find('input').simulate('change', { target: {value} });
		expect(wrapper.find('p').text().trim()).toBe(value);
	})

	test('should not post the category', () => {
			
		wrapper.find('form').simulate('submit', { preventDefault(){} })
		expect(setCategories).not.toHaveBeenCalled();
	})

	test('the input must be empty after adding a category', () => {

		const value = 'Hello World';

		wrapper.find('input').simulate('change', { target: {value} });
		wrapper.find('form').simulate('submit', { preventDefault(){} });

		expect(setCategories).toHaveBeenCalled();
		//expect(setCategories).toHaveBeenCalledTimes(1);
		expect(wrapper.find('input').prop('value')).toBe("");
	})
	
})
