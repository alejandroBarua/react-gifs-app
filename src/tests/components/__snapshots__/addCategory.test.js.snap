// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add category should be displayed correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Search Gifs..."
    type="text"
    value=""
  />
</form>
`;
